server/controllers/response.ts

```diff
@@ -78,3 +78,22 @@ export function processDeleteResult(req:Request, res: Response, next: NextFuncti
         res.redirect("/account");
     });
 }
+
+//Get Survey Result site
+export function displayResult(req:Request, res: Response, next: NextFunction): void
+{
+    const id = req.params.id;
+
+    ResponseM.find({ question: id }).then(data => {
+        const surveyResponse = data;
+        res.render("index", { title: "Survey Response", page: "surveyresponse", surveyResponses: surveyResponse});
+
+    });
+}
+
+// Get The list of results with a specific id
+export function getSurveyById(responsetitle: string, done: (err: any, res: ResponseM) => void): void
```

Comment:
I think you forgot the change the name of the function. Also, since you're trying to get a list of user responses, the signature of the callback function `done` should be `(err: any, res: ResponseM[]) => void`.

But the more important thing is, there should be a function doing the same thing in the same file. It's called `getAllResponse`. You should use that instead of creating a new function yourself.